docs/auth/README.md

```diff
@@ -0,0 +1,128 @@
+# Running Lighthouse on Authenticated Pages
+
+Default runs of Lighthouse load a page as a "new user", with no previous session or storage data. This means that pages requiring authenticated access do not work. There are multiple ways to run Lighthouse on an authenticated page - this document focuses on the most flexible approach ([Puppeteer](https://pptr.dev)), but mentions other approaches briefly at the end.
+
+If you just want to view the code, see [./example-lh-auth.js](example-lh-auth.js).
+
+## The Example Site
+
+There are two pages on the site:
+
+1. `/` - the homepage
+2. `/dashboard`
+
+The homepage shows the login form, but only to users that are not signed in.
+
+The dashboard shows a secret to users that are logged in, but shows an error to users that are not.
+
+The server responds with different HTML for each of these pages and session states, so there are four different pages that must have passable Lighthouse SEO scores.
+
+You can run this server locally if you like:
+
+```sh
+# be in root lighthouse directory
+yarn # install global project deps
+cd docs/auth
+yarn # install deps related to just this documentation
+yarn start # start the server on http://localhost:8000
+```
+
+## Puppeteer
+
+We can use Puppeteer - a browser automation tool - to manipulate a setup a session programatically.
+
+1. Launch a new browser.
+1. Navigate to the login page.
+1. Fill and submit the login form.
+1. Run Lighthouse using the same browser.
+
+First we launch Chrome:
+```js
+// This port will be used by Lighthouse later.
+const PORT = 8041;
+const browser = await puppeteer.launch({
+  args: [`--remote-debugging-port=${PORT}`],
+});
+```
+
+We navigate to the homepage, where the login form is:
+```js
+const page = await browser.newPage();
+await page.goto('http://localhost:8000');
+```
+
+Given a login form that looks like this:
+```html
+<form action="/login" method="post">
+  <label>
+    Email:
+    <input type="email" name="email">
+  </label>
+  <label>
+    Password:
+    <input type="password" name="password">
+  </label>
+  <input type="submit">
+</form>
+```
+
+We direct Puppeteer to fill and submit it:
+```js
+const emailInput = await page.$('input[type="email"]');
+await emailInput.type('admin@example.com');
+const passwordInput = await page.$('input[type="password"]');
+await passwordInput.type('password');
+const submitInput = await page.$('input[type="submit"]');
+await submitInput.press('Enter');
+await page.waitForNavigation();
```

Comment:
IIRC, @JoelEinbinder once told me I should be changing all my order of operations to

```js
await Promise.all([
  submitInput.click(),
  page.waitForNavigation(),
])
```

I'm not sure I ever noticed things going awry with the way you have it now though

Reply:
done. found this necessary w/ the eval submit approach